### PR TITLE
Slight updates to UDN documentation feature regarding DNS

### PR DESCRIPTION
**Pull request by stevsmit**
Version(s):
4.17+

Issue:
https://issues.redhat.com/browse/OCPBUGS-44205

Link to docs preview:
https://84445--ocpdocs-pr.netlify.app/openshift-enterprise/latest/networking/multiple_networks/understanding-user-defined-network.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

File: modules/nw-udn-limitations.adoc [hunk 1]
```diff
@@ -9,7 +9,10 @@
 While user-defined networks (UDN) offer highly customizable network configuration options, there are limitations that cluster administrators and developers should be aware of when implementing and managing these networks. Consider the following limitations before implementing a user-defined network.
 
 //Check on the removal of the DNS limitation for 4.18 or 4.17.z.
-* *DNS limitations*: DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
+* *DNS limitations*:
+** DNS for UDN pods is unsupported. The Kubernetes API (KAPI) and external DNS are supported.
```
Comment (kyrtapz) [inline modules/nw-udn-limitations.adoc:13]: This is a little confusing and I think the next sentence already covers the DNS behavior for UDN pods.

File: modules/nw-udn-limitations.adoc [hunk 2]
```diff
@@ -9,7 +9,10 @@
...
 
 //Check on the removal of the DNS limitation for 4.18 or 4.17.z.
-* *DNS limitations*: DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
+* *DNS limitations*:
+** DNS for UDN pods is unsupported. The Kubernetes API (KAPI) and external DNS are supported. 
+** DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
+** When a pod is assigned to a primary user-defined network, routes are set up to allow the pod to reach the Kubernetes API (KAPI) and DNS services on the cluster's default network. All other traffic is blocked by firewall rules on this network. Although KAPI and DNS are accessible by default, users might optionally configure access to other Kubernetes services on the cluster default network as well, however, the pod's IP shown by KAPI will continue to reflect the IP assigned in the cluster default network.
```
Comment (kyrtapz) [inline modules/nw-udn-limitations.adoc:15]: >  When a pod is assigned to a primary user-defined network, routes are set up to allow the pod to reach the Kubernetes API (KAPI) and DNS services on the cluster's default network. 

I would prefer if we avoid mentioning routes here and just say that KAPI and the DNS service are accessible.

> users might optionally configure access to other Kubernetes services on the cluster default network as well

We do not currently have support for that.

File: modules/nw-udn-limitations.adoc [hunk 3]
```diff
@@ -9,7 +9,10 @@
 While user-defined networks (UDN) offer highly customizable network configuration options, there are limitations that cluster administrators and developers should be aware of when implementing and managing these networks. Consider the following limitations before implementing a user-defined network.
 
 //Check on the removal of the DNS limitation for 4.18 or 4.17.z.
-* *DNS limitations*: DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
+* *DNS limitations*:
+** DNS for UDN pods is unsupported. The Kubernetes API (KAPI) and external DNS are supported. 
+** DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
```
Comment (kyrtapz) [inline modules/nw-udn-limitations.adoc:14]: I think it is valuable to add this from the enhancement:

> DNS lookups for services and external entities will function correctly.

https://github.com/openshift/enhancements/blob/master/enhancements/network/user-defined-network-segmentation.md#dns

File: modules/nw-udn-limitations.adoc [hunk 4]
```diff
@@ -9,7 +9,9 @@
 While user-defined networks (UDN) offer highly customizable network configuration options, there are limitations that cluster administrators and developers should be aware of when implementing and managing these networks. Consider the following limitations before implementing a user-defined network.
 
 //Check on the removal of the DNS limitation for 4.18 or 4.17.z.
-* *DNS limitations*: DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network.
+* *DNS limitations*:
+** DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups will not resolve to the pod's IP address on that user-defined network. However, DNS lookups for services and external entities will function as expected.
```
Comment (jeana-redhat) [inline modules/nw-udn-limitations.adoc:13]: Existing content, but can future tense be worked around here? 

```suggestion
** DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups cannot resolve to the pod's IP address on that user-defined network. However, DNS lookups for services and external entities function as expected.
```

or

```suggestion
** DNS lookups for pods resolve to the pod's IP address on the cluster default network. Even if a pod is part of a user-defined network, DNS lookups do not resolve to the pod's IP address on that user-defined network. However, DNS lookups for services and external entities function as expected.
```

If either messes up the semantics, feel free to ignore :)

**Review by jeana-redhat — COMMENTED** [review]
A thing to consider but this LGTM

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Comment by zhaozhanqi** [comment]
LGTM

**Pull request merged**